### PR TITLE
enable layout for rnn_state + enable additional attribute via kwargs …

### PR DESCRIPTION
**Pull request by sxjscience**
enable layout for rnn_state + enable additional attributes via kwargs in mx.sym.var

### CODE REVIEW

File: python/mxnet/symbol.py [hunk 1]
```diff
@@ -1152,6 +1153,11 @@ def var(name, attr=None, shape=None, lr_mult=None, wd_mult=None, dtype=None, ini
         attr['__dtype__'] = str(_DTYPE_NP_TO_MX[_numpy.dtype(dtype).type])
     if init is not None:
         attr['__init__'] = init.dumps()
+    for k, v in kwargs.items():
+        if k.startswith('__') and k.endswith('__'):
+            attr[k] = str(v)
+        else:
+            raise ValueError('attribute name=%s is not supported' %k)
```
Comment (piiswrong) [inline python/mxnet/symbol.py]: be clear, something like: "additional attributes must start and end with double underscores: __yourattr__"

File: python/mxnet/rnn/rnn_cell.py [hunk 2]
```diff
@@ -380,8 +380,9 @@ def __init__(self, num_hidden, prefix='lstm_', params=None, forget_bias=1.0):
         self._hB = self.params.get('h2h_bias')
 
     @property
-    def state_shape(self):
```
Comment (piiswrong) [inline python/mxnet/rnn/rnn_cell.py]: Keep the old interface.

File: python/mxnet/rnn/rnn_cell.py [hunk 3]
```diff
@@ -380,8 +380,9 @@ def __init__(self, num_hidden, prefix='lstm_', params=None, forget_bias=1.0):
...
-        return [(0, self._num_hidden), (0, self._num_hidden)]
+    def state_info(self):
+        return [{'shape': (0, self._num_hidden), 'layout': 'NC'},
```
Comment (piiswrong) [inline python/mxnet/rnn/rnn_cell.py]: shoudn't this be __layout__?

File: python/mxnet/rnn/rnn_cell.py (hunk 2 above)
Comment (sxjscience) [inline python/mxnet/rnn/rnn_cell.py]: Should I separate these two or merge into one property? 
Choice 1: `state_shape()` and `state_layout()`
Choice 2: `state_info()` only and the state_shape is defined based on the result of state_info

**Pull request merged**